crystal_diffusion/models/position_diffusion_lightning_model.py

```diff
@@ -27,7 +29,7 @@
 class PositionDiffusionParameters:
     """Position Diffusion parameters."""
 
-    score_network_parameters: MLPScoreNetworkParameters
+    score_network_parameters: typing.Union[MLPScoreNetworkParameters, MACEScoreNetworkParameters]
```

Comment:
Here I think you could just use `BaseScoreNetworkParameters` since both `MLP...` and `MACE...` are derived from it. 
Also, I would rename `BaseScoreNetworkParameters` to just`ScoreNetworkParameters`.